File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Effect/Combat/PowerBuffEffect.ts
import { FairyGUI } from "csharp";
import { View_PowerBuffEffect } from "../../Gen/View/ModTheSpire_Effect";
import { AbstractEffect } from "../AbstractEffect";

export class PowerBuffEffect extends AbstractEffect{
    public constructor(public X: number,public Y:number,public Msg:string) {
        super();
        let effect = <FairyGUI.GComponent>View_PowerBuffEffect.CreateInstance();
        effect.text = Msg;
        effect.SetPosition(X,Y,0);
        let transition = effect.GetTransition("t0");
        transition.Play(1,0,()=>{
            this.IsDone = true;
        });
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Effect/Combat/PowerBuffEffect.cs
using ModTheSpire.ModTheSpire_Effect;

namespace ModTheSpire.Effect.Combat
{
    public class PowerBuffEffect : AbstractEffect
    {
        private View_PowerBuffEffect mEffect;
        public PowerBuffEffect(float x, float y, string msg)
        {
            mEffect = View_PowerBuffEffect.CreateInstance();
            mEffect.text = msg;
            mEffect.SetPosition(x, y, 0);
            mEffect.GetTransition("t0").Play(1, 0, Finish);
        }

        public override void Dispose()
        {
            base.Dispose();
            mEffect.Dispose();
        }
    }
}